include/sample_rate_converter.h

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ * @brief Sample Rate Converter library header.
+ */
+
+#ifndef _SAMPLE_RATE_CONVERTER_H_
+#define _SAMPLE_RATE_CONVERTER_H_
+
+/**
+ * @defgroup sample_rate_converter Sample Rate Converter library
+ * @brief Implements functionality to increase or decrease the sample rate of a data array using
+ * the CMSIS DSP filtering library.
+ *
+ * @{
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/ring_buffer.h>
+#include <dsp/filtering_functions.h>
+
+/**
+ * Maximum size for the internal state buffers.
+ *
+ * The internal state buffer must for each context fulfill the following equations:
+ *	Interpolation:
+ *		number of filter taps + block size - 1
+ *	Decimation:
+ *		(number of filter taps / conversion ratio) + block size - 1
+ *
+ * The equation for interpolation is used as size as this gives the largest number.
+ */
+#define SAMPLE_RATE_CONVERTER_STATE_BUFFER_SIZE                                                    \
+	CONFIG_SAMPLE_RATE_CONVERTER_BLOCK_SIZE + CONFIG_SAMPLE_RATE_CONVERTER_MAX_FILTER_SIZE - 1
+
+/** Filter types supported by the sample rate converter */
+enum sample_rate_converter_filter {
+	SAMPLE_RATE_FILTER_SIMPLE,
+	SAMPLE_RATE_FILTER_SMALL,
+};
+
+/**
+ * To maintain filter requirements the input buffer must in some cases store two samples between
+ * each block processed.
+ */
+#define SAMPLE_RATE_CONVERTER_INPUT_BUFFER_NUMBER_OVERFLOW_SAMPLES 2
+
+/**
+ * To maintain filter requirements the output buffer must in some cases store six samples between
+ * each block processed. The output buffer must be able to store a full block in addition to these
+ * overflow samples.
+ */
+#define SAMPLE_RATE_CONVERTER_OUTPUT_BUFFER_NUMBER_OVERFLOW_SAMPLES 6
+
+#ifdef CONFIG_SAMPLE_RATE_CONVERTER_BIT_DEPTH_16
+#define SAMPLE_RATE_CONVERTER_INPUT_BUF_SIZE                                                       \
+	SAMPLE_RATE_CONVERTER_INPUT_BUFFER_NUMBER_OVERFLOW_SAMPLES * sizeof(uint16_t)
+#define SAMPLE_RATE_CONVERTER_RINGBUF_SIZE                                                         \
+	(CONFIG_SAMPLE_RATE_CONVERTER_BLOCK_SIZE +                                                 \
+	 SAMPLE_RATE_CONVERTER_OUTPUT_BUFFER_NUMBER_OVERFLOW_SAMPLES) *                            \
+		sizeof(uint16_t)
+#elif CONFIG_SAMPLE_RATE_CONVERTER_BIT_DEPTH_32
```

Comment:
Add an elif and error if the bit depth is 24

Reply:
These kconfigs are specific to the sample rate converter library, so only kconfigs for 32 and 16 exists. So if I understand you correctly, if the user tries to configure the audio project to use the sample rate converter with a 24 bit depth system, we should raise an error at the audio application level